flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/plan/schema/GenericRelDataType.scala

```diff
@@ -55,4 +57,8 @@ class GenericRelDataType(
   override def hashCode(): Int = {
     genericType.hashCode()
   }
+
+  override def generateTypeString(sb: lang.StringBuilder, withDetail: Boolean): Unit = {
```

Comment:
Now we delete `ArrayRelDataType` and use `ArraySqlType` and etc...
NOTE: `equals` of `ArraySqlType` is:
```
@Override public boolean equals(Object obj) {
    if (obj instanceof RelDataTypeImpl) {
      final RelDataTypeImpl that = (RelDataTypeImpl) obj;
      return this.digest.equals(that.digest);
    }
    return false;
  }
```
So we must need override `generateTypeString` all calcite extended type.

Reply:
Put the reason to the method comment